soa-derive-internal/src/iter.rs

```diff
@@ -136,6 +136,17 @@ pub fn derive(input: &Struct) -> Tokens {
                 }
             }
 
+            impl<'a> IntoIterator for &'a #slice_name<'a> {
```

Comment:
Do we actually want the reference and the struct lifetime to be the same ? This could be `impl<'a, 'b> IntoIterator for &'a #slice_name<'b>` instead. I could see such case occurs in code like this

```rust
let slice =  cheese_vec.slice();
let ref = &slice;
// Do stuff with the reference
```
Or like this
```rust
let slice =  cheese_vec.slice();
do_stuff_with_reference(&slice);
```

Reply:
Changed. Thank you for the review.